core/templates/dev/head/pages/exploration_editor/editor_tab/TrainingDataEditorPanelService.js

```diff
@@ -0,0 +1,165 @@
+              var _state = ExplorationStatesService.getState(_stateName);
+              var answerGroupIndex = (
+                ResponsesService.getActiveAnswerGroupIndex());
+
+              $scope.stateContent = _state.content.getHtml();
+              $scope.inputTemplate = (
+                ExplorationHtmlFormatterService.getInteractionHtml(
+                  stateInteractionIdService.savedMemento,
+                  stateCustomizationArgsService.savedMemento,
+                  false, 'testInteractionInput'));
+              $scope.trainingData = [];
+
+              var _rebuildTrainingData = function() {
+                $scope.trainingData = [];
+                TrainingDataService.getTrainingDataOfAnswerGroup(
+                  answerGroupIndex).forEach(function(answer) {
+                  answerTemplate = (
+                    ExplorationHtmlFormatterService.getAnswerHtml(
+                      answer, stateInteractionIdService.savedMemento,
+                      stateCustomizationArgsService.savedMemento));
+                  $scope.trainingData.push({
+                    answer: answer,
+                    answerTemplate: answerTemplate
+                  });
+                });
+              };
+
+              $scope.init = function() {
+                _rebuildTrainingData();
+                $scope.newAnswerIsResolved = false;
+                $scope.answerSuccessfullyAdded = false;
+              };
+
+              $scope.removeAnswerFromTrainingData = function(answerIndex) {
+                answer = $scope.trainingData[answerIndex].answer;
+                TrainingDataService.removeAnswerFromAnswerGroupTrainingData(
+                  answer, answerGroupIndex);
+                $scope.trainingData.splice(answerIndex, 1);
+              };
+
+              $scope.exit = function() {
+                $uibModalInstance.close();
+              };
+
+              $scope.submitAnswer = function(newAnswer) {
+                $scope.newAnswerIsResolved = false;
+                $scope.answerSuccessfullyAdded = false;
+
+                var interactionId = stateInteractionIdService.savedMemento;
+
+                var rulesServiceName =
+                  AngularNameService.getNameOfInteractionRulesService(
+                    interactionId);
+
+                // Inject RulesService dynamically.
+                var rulesService = $injector.get(rulesServiceName);
+
+                var newAnswerTemplate = (
+                  ExplorationHtmlFormatterService.getAnswerHtml(
+                    newAnswer, stateInteractionIdService.savedMemento,
+                    stateCustomizationArgsService.savedMemento));
+
+                var classificationResult = (
+                  AnswerClassificationService.getMatchingClassificationResult(
+                    _explorationId, _stateName, _state,
+                    newAnswer, rulesService));
+                var newAnswerFeedbackHtml = 'Nothing';
```

Comment:
This seems weird. Shouldn't it be null or the empty string, and the UI displays it as "Nothing" or "[No feedback specified]" if necessary?

Reply:
I don't know. This is how it was in previous interface and so I kept this part as it is.

Reply:
Ah ok. Let's change it to what I suggested? Semantically that would be better, I think, rather than using a sentinel value of 'Nothing'.

Reply:
Done.